crates/apollo-mcp-server/src/runtime/logging.rs

```diff
@@ -52,8 +50,18 @@ impl Default for Logging {
     }
 }
 
+type LoggingLayerResult = (
+    Layer<
+        tracing_subscriber::Registry,
+        tracing_subscriber::fmt::format::DefaultFields,
+        tracing_subscriber::fmt::format::Format,
+        BoxMakeWriter,
+    >,
+    Option<tracing_appender::non_blocking::WorkerGuard>,
+);
+
 impl Logging {
-    pub fn setup(config: &Config) -> Result<Option<WorkerGuard>, anyhow::Error> {
+    pub fn env_filter(config: &Config) -> Result<EnvFilter, anyhow::Error> {
```

Comment:
Would it make sense to pass only `config.logging` to this function, since that's the only part of the config it actually uses?